cluster/src/main/java/io/atomix/cluster/messaging/impl/DefaultClusterMessagingService.java

```diff
@@ -133,13 +134,17 @@ public DefaultClusterMessagingService(ClusterService cluster, MessagingService m
 
   private CompletableFuture<Void> doUnicast(String subject, byte[] payload, NodeId toNodeId) {
     Node node = cluster.getNode(toNodeId);
-    checkArgument(node != null, "Unknown nodeId: %s", toNodeId);
+    if (node == null) {
+      return Futures.exceptionalFuture(new ConnectException());
+    }
     return messagingService.sendAsync(node.endpoint(), subject, payload);
   }
 
   private CompletableFuture<byte[]> sendAndReceive(String subject, byte[] payload, NodeId toNodeId) {
     Node node = cluster.getNode(toNodeId);
-    checkArgument(node != null, "Unknown nodeId: %s", toNodeId);
+    if (node == null) {
+      return Futures.exceptionalFuture(new ConnectException());
```

Comment:
Could omit the stacktrace for performance?

Reply:
yep good idea

Reply:
fixed